### PR TITLE
:seedling: Use basic-checks image in generate and gomod tests

### PR DESCRIPTION
**Pull request by mquhuy**
As a continuation of https://github.com/metal3-io/project-infra/pull/797, we use basic-checks image in local tests too for BMO.

### CODE REVIEW

**Review by tuminoid — COMMENTED** [review]
/lgtm

These work, but it appears we have some Docker related mount option issues in some of the hack scripts, that prevent returning success right now when run locally. I'll take a look at those.

/hold
Until project-infra PR is merged.

**Comment by tuminoid** [comment]
Also, let's merge this first https://github.com/metal3-io/baremetal-operator/pull/1796 so this can be tested locally. Changing the docker image here doesn't mean anything from CI perspective, so it needs a manual test.

**Review by Rozzii — COMMENTED** [review]
/lgtm

File: hack/generate.sh
Comment (tuminoid) [inline hack/generate.sh:35]:
```suggestion
        --pull=always \
        quay.io/metal3-io/basic-checks:golang-1.22 \
```
Otherwise locally we never will get updates made to that image after the first pull is done. For golang image it is not meaningful as its just bug fixes, but here if we add a new package here, it wouldn't show up locally unless manually forcing cleanup/repull.

File: hack/gomod.sh
Comment (tuminoid) [inline hack/gomod.sh:44]: same here

**Review by Rozzii — COMMENTED** [review]
/lgtm

**Review by tuminoid — COMMENTED** [review]
/unhold
/lgtm

**Review by dtantsur — APPROVED** [review]

**Pull request merged**
